packages/@expo/image-utils/src/sharp.ts

```diff
@@ -104,35 +112,53 @@ let _sharpBin: string | null = null;
 let _sharpInstance: any | null = null;
 
 async function findSharpBinAsync(): Promise<string> {
-  if (_sharpBin) {
-    return _sharpBin;
-  }
+  if (_sharpBin) return _sharpBin;
+
   try {
-    const sharpCliPackage = require('sharp-cli/package.json');
-    const libVipsVersion = require('sharp').versions.vips;
+    const sharpCliPackagePath =
+      resolveGlobal.silent('sharp-cli/package.json') ??
+      require.resolve('sharp-cli/package.json', {
+        paths: require.resolve.paths('sharp-cli') ?? undefined,
+      });
+
+    const sharpCliPackage = require(sharpCliPackagePath);
+
+    _sharpInstance = sharpCliPackagePath
+      ? require(resolveFrom(sharpCliPackagePath, 'sharp'))
+      : null;
+
     if (
-      sharpCliPackage &&
+      sharpCliPackagePath &&
       semver.satisfies(sharpCliPackage.version, SHARP_REQUIRED_VERSION) &&
       typeof sharpCliPackage.bin.sharp === 'string' &&
-      typeof libVipsVersion === 'string'
+      typeof _sharpInstance?.versions?.vips === 'string'
     ) {
-      _sharpBin = require.resolve(`sharp-cli/${sharpCliPackage.bin.sharp}`);
-      return _sharpBin;
+      _sharpBin = path.join(path.dirname(sharpCliPackagePath), sharpCliPackage.bin.sharp);
+    } else {
+      _sharpInstance = null;
+    }
+  } catch (error) {
+    _sharpBin = null;
+    _sharpInstance = null;
+
+    // `sharp-cli` and/or `sharp` modules could not be found, falling back to global binary only
+    if (env.EXPO_IMAGE_UTILS_DEBUG) {
+      console.warn('Sharp could not be loaded, reason:', error);
     }
```

Comment:
left over; rather than `debug`?

Reply:
No, it follows the already existing pattern for this package https://github.com/expo/expo/blob/3590b42e8e687c4da83331642b3b4d92981c0be2/packages/%40expo/image-utils/src/Image.ts#L114-L118